code/crates/sync/src/scoring/ema.rs

```diff
@@ -0,0 +1,107 @@
+use std::time::Duration;
+
+use malachitebft_peer::PeerId;
+
+use super::{Score, ScoringStrategy, SyncResult};
+
+/// Exponential Moving Average scoring strategy
+#[derive(Copy, Clone, Debug)]
+pub struct ExponentialMovingAverage {
+    /// Learning rate for successful responses
+    pub alpha_success: f64,
+
+    /// Learning rate for timeouts
+    pub alpha_timeout: f64,
+
+    /// Learning rate for failures
+    pub alpha_failure: f64,
+
+    /// Threshold for slow responses.
+    ///
+    /// This should typically be smaller than both the expected
```

Comment:
Should we validate this value, so we could at least to emit a warning? I'm not sure though where is best to check for that, because this condition is specific to this strategy, then we would need to add block time and sync request timeout to the strategy constructor.